includes/tuque_wrapper.inc

```diff
@@ -436,6 +436,88 @@ class IslandoraFedoraApiM extends FedoraApiM {
     }
   }
 
+  /**
+   * Wraps purgeDatastream for semaphore locking.
+   *
+   * @see FedoraApiM::purgeDatastream
+   */
+  public function purgeDatastream($pid, $dsid, $params = array()) {
+    return $this->callParentWithLocking('purgeDatastream', $pid, $pid, $dsid, $params);
+  }
+
+  /**
+   * Wraps ingest for semaphore locking.
+   *
+   * @see FedoraApiM::ingest
+   */
+  public function ingest($params = array()) {
```

Comment:
@willtp87 or @jordandukart. Just a question:  Do we use `::ingest` directly/currently on the islandora side?  I can't find any reference to that nor the parent method.